tests/Microsoft.Identity.Test.Unit/AppConfigTests/ConfidentialClientApplicationBuilderTests.cs

```diff
@@ -94,6 +94,64 @@ public void TestBuildWithInstanceWithTrailingSlash()
             TestBuildConfidentialClientFromOptions(options);
         }
 
+        [TestMethod]
+        public void CacheSynchronization()
+        {
```

Comment:
nit: you could just pass in the EnableCacheSycnro as a parameter to the test and then remove some of the duplicate code...just a thought.